### DISCUSSION TITLE
Welcome to Downloader Discussions!

### DISCUSSION THREAD

**Discussion by bezzad** [bezzad/Downloader#25] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by GihanSoft** [comment]
Can you just change license to anything else like apache or mit?
gpl-3 actually prevent using this lib in many cases.

**Reply by ghost1372** [reply]
I agree too
MIT is better

**Comment by bezzad** [comment]
Ok, no problem. I will change it to the MIT license.
Feel free to use it in any type of software free or non-free.

**Reply by GihanSoft** [reply]
For real? Many tanks.

**Comment by GihanSoft** [comment]
Hi @bezzad. About that API I designed, what kind of tests it need? can you guide me a little bit?

**Comment by bezzad** [comment]
@GihanSoft, Just write a unit test for each method of your classes. If you want to know how to test, so you can see the `Downloader.Test` project to know better what can you do. 
Let me just say that you do not need to test how to download because this part has been tested. Just test your business of classes. Some places are very sensitive and you can write unit tests more to make sure it works properly. For example: How can you make sure that if the user just sets the folder path (e.x `.WithFolder("/Path/To/File")`) and the Downloader can save the file with the real name from the download URL.
